client/config.c

```diff
@@ -26,6 +29,52 @@ TDNFConfGetRpmVerbosity(
     return nLogLevel;
 }
 
+static uint32_t TDNFParseOSInfo(PTDNF_CONF pConf)
+{
+    char temp[USERAGENTHEADERMAXLENGTH];
+    char *name = NULL;
+    char *version = NULL;
+    uint32_t dwError = 0;
+    FILE *file = NULL;
+
+    file = fopen(OSCONFFILE, "r");
+
+    if (!file) {
+        dwError = errno;
+        BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+    }
+
+    dwError = TDNFAllocateMemory(10, sizeof(char), (void**)&name);
```

Comment:
Don't pre-allocate, parse the strings into the temp buffer first, then use `TDNFAllocateString()` to allocate the exact amount you need.